### ISSUE TITLE
remove columns (genes) with more than X% missing values before selecting most variable genes

### ISSUE THREAD

**Issue by ngehlenborg** [Caleydo/caleydo#1541]
See #1540 for an example where some microRNAs seem to have missing values in almost every sample.

I will investigate what Firehose is doing about this.

**Comment by alexsb** [comment]
What would be a sensible X? 40%? I guess if there are many missing values the trust in the rest is also low?

**Comment by ngehlenborg** [comment]
I have just had another look at the Firehose code used to do the scaling, and if I am interpreting this correctly, they are removing the gene if there are more than 80% NAs.

``` R
exp <- read.data(expfile)

    # Remove up to 2 leading non-numeric columns.
    if (all(is.na(as.numeric(exp[,1])))) {
        exp=exp[, -1,drop=F]
        if (all(is.na(as.numeric(exp[,1])))) {
            exp=exp[, -1,drop=F]
        }
    }

    exp[exp == -Inf] <- NA
    temp <- apply(exp,2,as.numeric)
    row.names(temp) <- row.names(exp)
    temp <- scale(t(temp), center = TRUE, scale = FALSE)
    exp <- t(temp)

    colnames(exp) <- gsub("\\.","-",colnames(exp))

    na.number  <- apply(exp,2,function(x) sum(is.na(x)))
    t <- which(na.number > 0.8*dim(exp)[1])

    if(length(t) > 0 ) {exp <- exp[,-t]}
```

**Comment by alexsb** [comment]
While this is implemented, this does not fix or address situations in #1540. In this case, no sampling is done at all, as it's only 450 or so dimensions in total. 

For the full matrices we currently consider there seems to be not a single case where this plays a role.
